4 tests erros when validating build in rpm spec

when I activated testing the build with

```
PATH=.:$PATH py.test tests/
```

I got

> 4 failed, 169 passed in 3.06 seconds

here is the full build logs

https://paste.fedoraproject.org/414308/22152031/

```
============================= test session starts ==============================
platform linux2 -- Python 2.7.12, pytest-2.9.2, py-1.4.31, pluggy-0.3.1
rootdir: /builddir/build/BUILD/dumb-init-1.1.3, inifile: pytest.ini
collected 173 items
tests/child_processes_test.py ........................
tests/cli_test.py ......................................................................
tests/exit_status_test.py ..................................F...F...F...F.
tests/proxies_signals_test.py ..........................
tests/shell_background_test.py ....
tests/tty_test.py .
=================================== FAILURES ===================================
_________________ test_exit_status_terminated_by_signal[1-1-3] _________________
signal = 3
    @pytest.mark.parametrize('signal', [
        signal.SIGTERM,
        signal.SIGINT,
        signal.SIGQUIT,
        signal.SIGKILL,
    ])
    @pytest.mark.usefixtures('both_debug_modes', 'both_setsid_modes')
    def test_exit_status_terminated_by_signal(signal):
        """dumb-init should exit with status 128 + signal when the child process is
        terminated by a signal.
        """
        proc = Popen(('dumb-init', 'sh', '-c', 'kill -{0} $$'.format(signal)))
        proc.wait()
>       assert proc.returncode == 128 + signal
E       assert 0 == (128 + 3)
E        +  where 0 = <subprocess.Popen object at 0x7feaa59c0790>.returncode
tests/exit_status_test.py:31: AssertionError
----------------------------- Captured stderr call -----------------------------
[dumb-init] Running in debug mode.
[dumb-init] Child spawned with PID 5951.
[dumb-init] setsid complete.
[dumb-init] Received signal 17.
[dumb-init] A child with PID 5951 exited with exit status 0.
[dumb-init] Forwarded signal 15 to children.
[dumb-init] Child exited with status 0. Goodbye.
_________________ test_exit_status_terminated_by_signal[1-0-3] _________________
signal = 3
    @pytest.mark.parametrize('signal', [
        signal.SIGTERM,
        signal.SIGINT,
        signal.SIGQUIT,
        signal.SIGKILL,
    ])
    @pytest.mark.usefixtures('both_debug_modes', 'both_setsid_modes')
    def test_exit_status_terminated_by_signal(signal):
        """dumb-init should exit with status 128 + signal when the child process is
        terminated by a signal.
        """
        proc = Popen(('dumb-init', 'sh', '-c', 'kill -{0} $$'.format(signal)))
        proc.wait()
>       assert proc.returncode == 128 + signal
E       assert 0 == (128 + 3)
E        +  where 0 = <subprocess.Popen object at 0x7feaa59d6190>.returncode
tests/exit_status_test.py:31: AssertionError
----------------------------- Captured stderr call -----------------------------
[dumb-init] Running in debug mode.
[dumb-init] Not running in setsid mode.
[dumb-init] Child spawned with PID 5959.
[dumb-init] Received signal 17.
[dumb-init] A child with PID 5959 exited with exit status 0.
[dumb-init] Forwarded signal 15 to children.
[dumb-init] Child exited with status 0. Goodbye.
_________________ test_exit_status_terminated_by_signal[0-1-3] _________________
signal = 3
    @pytest.mark.parametrize('signal', [
        signal.SIGTERM,
        signal.SIGINT,
        signal.SIGQUIT,
        signal.SIGKILL,
    ])
    @pytest.mark.usefixtures('both_debug_modes', 'both_setsid_modes')
    def test_exit_status_terminated_by_signal(signal):
        """dumb-init should exit with status 128 + signal when the child process is
        terminated by a signal.
        """
        proc = Popen(('dumb-init', 'sh', '-c', 'kill -{0} $$'.format(signal)))
        proc.wait()
>       assert proc.returncode == 128 + signal
E       assert 0 == (128 + 3)
E        +  where 0 = <subprocess.Popen object at 0x7feaa59c0850>.returncode
tests/exit_status_test.py:31: AssertionError
_________________ test_exit_status_terminated_by_signal[0-0-3] _________________
signal = 3
    @pytest.mark.parametrize('signal', [
        signal.SIGTERM,
        signal.SIGINT,
        signal.SIGQUIT,
        signal.SIGKILL,
    ])
    @pytest.mark.usefixtures('both_debug_modes', 'both_setsid_modes')
    def test_exit_status_terminated_by_signal(signal):
        """dumb-init should exit with status 128 + signal when the child process is
        terminated by a signal.
        """
        proc = Popen(('dumb-init', 'sh', '-c', 'kill -{0} $$'.format(signal)))
        proc.wait()
>       assert proc.returncode == 128 + signal
E       assert 0 == (128 + 3)
E        +  where 0 = <subprocess.Popen object at 0x7feaa59d8b10>.returncode
tests/exit_status_test.py:31: AssertionError
===================== 4 failed, 169 passed in 3.06 seconds =====================
```

I can reproduce this:

```
ckuehl@debian:~$ docker run -ti fedora bash
[root@6ffe37019bfe /]# dnf install git gcc make glibc-static python3-pytest python3-mock
[root@6ffe37019bfe /]# git clone https://github.com/Yelp/dumb-init
Cloning into 'dumb-init'...
remote: Counting objects: 810, done.
remote: Compressing objects: 100% (10/10), done.
remote: Total 810 (delta 2), reused 0 (delta 0), pack-reused 800
Receiving objects: 100% (810/810), 164.13 KiB | 0 bytes/s, done.
Resolving deltas: 100% (462/462), done.
Checking connectivity... done.
[root@6ffe37019bfe /]# cd dumb-init
[root@6ffe37019bfe dumb-init]# make
cc -std=gnu99 -static -s -Wall -Werror -O3 -o dumb-init dumb-init.c
[root@6ffe37019bfe dumb-init]# PATH=.:$PATH py.test-3 tests/
================================== test session starts ===================================
platform linux -- Python 3.5.1, pytest-2.9.2, py-1.4.31, pluggy-0.3.1
rootdir: /dumb-init, inifile: pytest.ini
collected 173 items

tests/child_processes_test.py ........................
tests/cli_test.py ......................................................................
tests/exit_status_test.py ..................................F...F...F...F.
tests/proxies_signals_test.py ..........................
tests/shell_background_test.py ....
tests/tty_test.py .

======================================== FAILURES ========================================
_______________ test_exit_status_terminated_by_signal[1-1-Signals.SIGQUIT] _______________

signal = <Signals.SIGQUIT: 3>

    @pytest.mark.parametrize('signal', [
        signal.SIGTERM,
        signal.SIGINT,
        signal.SIGQUIT,
        signal.SIGKILL,
    ])
    @pytest.mark.usefixtures('both_debug_modes', 'both_setsid_modes')
    def test_exit_status_terminated_by_signal(signal):
        """dumb-init should exit with status 128 + signal when the child process is
        terminated by a signal.
        """
        proc = Popen(('dumb-init', 'sh', '-c', 'kill -{0} $$'.format(signal)))
        proc.wait()
>       assert proc.returncode == 128 + signal
E       assert 0 == (128 + <Signals.SIGQUIT: 3>)
E        +  where 0 = <subprocess.Popen object at 0x7f203d940550>.returncode

tests/exit_status_test.py:31: AssertionError
---------------------------------- Captured stderr call ----------------------------------
[dumb-init] Running in debug mode.
[dumb-init] Child spawned with PID 633.
[dumb-init] setsid complete.
[dumb-init] Received signal 17.
[dumb-init] A child with PID 633 exited with exit status 0.
[dumb-init] Forwarded signal 15 to children.
[dumb-init] Child exited with status 0. Goodbye.
_______________ test_exit_status_terminated_by_signal[1-0-Signals.SIGQUIT] _______________

signal = <Signals.SIGQUIT: 3>

    @pytest.mark.parametrize('signal', [
        signal.SIGTERM,
        signal.SIGINT,
        signal.SIGQUIT,
        signal.SIGKILL,
    ])
    @pytest.mark.usefixtures('both_debug_modes', 'both_setsid_modes')
    def test_exit_status_terminated_by_signal(signal):
        """dumb-init should exit with status 128 + signal when the child process is
        terminated by a signal.
        """
        proc = Popen(('dumb-init', 'sh', '-c', 'kill -{0} $$'.format(signal)))
        proc.wait()
>       assert proc.returncode == 128 + signal
E       assert 0 == (128 + <Signals.SIGQUIT: 3>)
E        +  where 0 = <subprocess.Popen object at 0x7f203d8cb320>.returncode

tests/exit_status_test.py:31: AssertionError
---------------------------------- Captured stderr call ----------------------------------
[dumb-init] Running in debug mode.
[dumb-init] Not running in setsid mode.
[dumb-init] Child spawned with PID 641.
[dumb-init] Received signal 17.
[dumb-init] A child with PID 641 exited with exit status 0.
[dumb-init] Forwarded signal 15 to children.
[dumb-init] Child exited with status 0. Goodbye.
_______________ test_exit_status_terminated_by_signal[0-1-Signals.SIGQUIT] _______________

signal = <Signals.SIGQUIT: 3>

    @pytest.mark.parametrize('signal', [
        signal.SIGTERM,
        signal.SIGINT,
        signal.SIGQUIT,
        signal.SIGKILL,
    ])
    @pytest.mark.usefixtures('both_debug_modes', 'both_setsid_modes')
    def test_exit_status_terminated_by_signal(signal):
        """dumb-init should exit with status 128 + signal when the child process is
        terminated by a signal.
        """
        proc = Popen(('dumb-init', 'sh', '-c', 'kill -{0} $$'.format(signal)))
        proc.wait()
>       assert proc.returncode == 128 + signal
E       assert 0 == (128 + <Signals.SIGQUIT: 3>)
E        +  where 0 = <subprocess.Popen object at 0x7f203d96b780>.returncode

tests/exit_status_test.py:31: AssertionError
_______________ test_exit_status_terminated_by_signal[0-0-Signals.SIGQUIT] _______________

signal = <Signals.SIGQUIT: 3>

    @pytest.mark.parametrize('signal', [
        signal.SIGTERM,
        signal.SIGINT,
        signal.SIGQUIT,
        signal.SIGKILL,
    ])
    @pytest.mark.usefixtures('both_debug_modes', 'both_setsid_modes')
    def test_exit_status_terminated_by_signal(signal):
        """dumb-init should exit with status 128 + signal when the child process is
        terminated by a signal.
        """
        proc = Popen(('dumb-init', 'sh', '-c', 'kill -{0} $$'.format(signal)))
        proc.wait()
>       assert proc.returncode == 128 + signal
E       assert 0 == (128 + <Signals.SIGQUIT: 3>)
E        +  where 0 = <subprocess.Popen object at 0x7f203d8cbd30>.returncode

tests/exit_status_test.py:31: AssertionError
========================== 4 failed, 169 passed in 5.26 seconds ==========================
```

--- a/tests/exit_status_test.py
+++ b/tests/exit_status_test.py
@@ -1,3 +1,4 @@
+import distutils.spawn
 import signal
 from subprocess import Popen
 
@@ -26,6 +27,11 @@ def test_exit_status_terminated_by_signal(signal):
     """dumb-init should exit with status 128 + signal when the child process is
     terminated by a signal.
     """
-    proc = Popen(('dumb-init', 'sh', '-c', 'kill -{0} $$'.format(signal)))
+    # We need to make sure not to use the built-in kill (if on Bash):
+    # https://github.com/Yelp/dumb-init/issues/115
+    proc = Popen(('dumb-init', 'sh', '-c', '{0} -{1} $$'.format(
+        distutils.spawn.find_executable('kill'),
+        signal,
+    )))
     proc.wait()
     assert proc.returncode == 128 + signal

--- a/tests/exit_status_test.py
+++ b/tests/exit_status_test.py
@@ -1,5 +1,5 @@
-import distutils.spawn
 import signal
+import sys
 from subprocess import Popen
 
 import pytest
@@ -18,7 +18,7 @@ def test_exit_status_regular_exit(exit_status):
 
 @pytest.mark.parametrize('signal', [
     signal.SIGTERM,
-    signal.SIGINT,
+    signal.SIGHUP,
     signal.SIGQUIT,
     signal.SIGKILL,
 ])
@@ -27,10 +27,9 @@ def test_exit_status_terminated_by_signal(signal):
     """dumb-init should exit with status 128 + signal when the child process is
     terminated by a signal.
     """
-    # We need to make sure not to use the built-in kill (if on Bash):
+    # We use Python because sh is "dash" on Debian and "bash" on others.
     # https://github.com/Yelp/dumb-init/issues/115
-    proc = Popen(('dumb-init', 'sh', '-c', '{0} -{1} $$'.format(
-        distutils.spawn.find_executable('kill'),
+    proc = Popen(('dumb-init', sys.executable, '-c', 'import os; os.kill(os.getpid(), {0})'.format(
         signal,
     )))
     proc.wait()

